pkg/reconciler/reconciler.go

```diff
@@ -257,13 +257,15 @@ func Run(opts Options) {
 		}
 	}
 
-	nsControllerState := namespacecontroller.NewState()
+	var nsControllerState *namespacecontroller.State
 	if opts.ReconcilerScope == declared.RootReconciler {
 		rootOpts := &parse.RootOptions{
 			SourceFormat:             opts.SourceFormat,
 			NamespaceStrategy:        opts.NamespaceStrategy,
 			DynamicNSSelectorEnabled: opts.DynamicNSSelectorEnabled,
-			NSControllerState:        nsControllerState,
+		}
```

Comment:
Does the state need to be initialized for RepoSyncs or just RootSyncs?

Reply:
Just RootSyncs because RepoSyncs can't manage NamespaceSelector.

Reply:
Gotcha. Can you add an inline comment here to explain?
I understand it now, I just want to make sure it's clear for someone without context why we're only setting it in this 2nd level if statement.

Reply:
done, also added a comment to the field in `RootOptions`.